src/esp/assets/ResourceManager.cpp

```diff
@@ -3522,6 +3522,41 @@ void ResourceManager::setLightSetup(gfx::LightSetup setup,
                      Mn::ResourcePolicy::Manual);
 }
 
+int ResourceManager::registerRigInstance(gfx::Rig&& rig) {
```

Comment:
Yes, I think that this would make sense. `ResourceManager` is already bloated enough and the lifetime of these instance would be better controlled.

>Possibly related: BatchReplayRenderer doesn't (and never should) have a ResourceManager.
Agreed. I'm on the fence whether we should do that refactor here, or along with the skinned mesh integration with the batch renderer.